iceoryx2-bb/lock-free/src/spsc/index_queue.rs

```diff
@@ -42,16 +42,21 @@
 //! }
 //! ```
 
-use core::{alloc::Layout, cell::UnsafeCell, fmt::Debug, sync::atomic::Ordering};
+#[cfg(all(test, loom, feature = "std"))]
```

Comment:
If the type is not coming from `std`, perhaps the `std` feature gate could  be removed. The check for the `loom` environment would be enough. It's probably functionally the same, but  it seems semantically inaccurate.

Does the loom setup work if removing the `std` feature gate?

Reply:
> Does the loom setup work if removing the std feature gate?

Yes, it should work. Let me update the code so that the std feature gate is applied only to the parts that require std.